File: code/jeffrey/Python/lab25_atm.py
class ATM:
    
    # initializes with balance and sets interest rate default
    def __init__ (self, balance = 0, int_rate = .001):
        self.balance= balance
        self.int_rate= int_rate
        self.trans = []

    # returns the account balance
    def check_balance(self):
        return f'{self.balance}'

    # deposits the given amount in the account
    def deposit(self, amount):
        self.balance += amount
        self.trans.append(f'${amount} deposited')
        return self.balance
        
    # returns true if the withdrawn amount won't put the account in the negative 
    def __check_withdrawal(self, amount):
        if (self.balance - amount) >= 0:
            return True
        else:
            return False

    # withdraws the amount from the account and returns it
    def withdraw(self, amount):
        if self.__check_withdrawal(amount):
            self.balance -= amount
            self.trans.append(f'${amount} withdrawn')
            return self.balance
        else:
            return f"Error: Not enough money to withdraw."

    # returns the amount of interest calculated on the account
    def calc_interest(self):
        return (self.balance * self.int_rate)

    # prints out a history of transactions
    def print_transactions(self):
        # for i in range(len(self.trans)):
        #     print(self.trans[i])
        if not self.trans:
            return str('No transactions yet')
        else:
            return ', '.join(self.trans)

cash = ATM(500)
flag = True
print("Thank you for using VERYSECUREDEFINITELYNOTFRAUD ATM\nWhat action would you like to do:\n1 for Deposit\n2 for Withdraw\n3 for Check Balance\n4 for Transaction History\n5 for Finish")

while flag:
    user_input = int(input('>'))

    if user_input == 1:
        try:
            deposit_amount = int(input("How much would you like to deposit: $"))
            cash.deposit(deposit_amount)
        except:
            print('Enter a valid amount.')
    elif user_input == 2:
        withdraw_amount = int(input("How much would you like to withdraw: $"))
        try:
            if "Error" in cash.withdraw(withdraw_amount):
                print(str("Not enough funds for that withdrawal amount."))
        except:
            print(f'Account debited by ${withdraw_amount}.')
    elif user_input == 3:
        print(f'***Your balance on this account is ${cash.check_balance()}***')
    elif user_input ==4:
        print(cash.print_transactions())
    elif user_input ==5:
        flag = False
    

File: code/jeffrey/JavaScript/lab25_atm.js
let cash;
let flag;

class Atm {

    // initializes with balance and sets interest rate default
  constructor (balance = 0, int_rate = .001, trans = []) {
      this.balance = balance;
      this.int_rate = int_rate;
      this.trans = trans;
  }

  // returns the account balance
  checkBalance() {
      return this.balance;
  }

  // deposits the given amount in the account
  deposit(amount) {
      this.balance += amount;
      this.trans.push(`$${amount} deposited`);
      return this.balance;
  }

  // returns true if the withdrawn amount won't put the account in the negative
  checkWithdrawal(amount){
      if ((this.balance - amount) >= 0) {
          return true;
      } else {
          return false;
        }
  }

  // withdraws the amount from the account and returns it
  withdraw(amount) {
      if (this.checkWithdrawal(amount) === true) {
        this.balance -= amount;
        this.trans.push(`$${amount} withdrawn`);
        return this.balance;
      } else {
        return ("Error: Not enough money to withdraw that amount.");
      }
  }

  // returns the amount of interest calculated on the account
  calcinterest() {
      return (this.balance * this.int_rate);
  }

  // prints out a history of transactions
  printTransactions() {
    if (this.trans.length == 0) {
      return ("No transactions yet.");
    } else {
      return (`Transactions: ${this.trans.join(", ")}`);
    }
  }
}

cash = new Atm(500);
flag = true;
alert("Thank you for using VERYSECUREDEFINITELYNOTFRAUD ATM\nWhat action would you like to do:\n1 for Deposit\n2 for Withdraw\n3 for Check Balance\n4 for Transaction History\n5 for Finish");

while (flag) {
  user_input = parseInt(prompt('>'));

  if (user_input === 1) {
    amount = parseFloat(prompt("How much would you like to deposit: $"));
    alert(`Account credited with $${amount}.`);
    cash.deposit(amount);
  } else if (user_input === 2) {
      amount = parseFloat(prompt("How much would you like to withdraw: $"));
      if (cash.checkWithdrawal(amount) === false){
        alert("Not enough funds for that withdrawal amount.");
      } else {
        cash.withdraw(amount);
        alert(`Account debited by $${amount}.`)}
  } else if (user_input === 3) {
      alert(`***Your balance on this account is ${cash.checkBalance()}***`);
  } else if (user_input === 4) {
      alert(cash.printTransactions());
  } else if (user_input === 5) {
      flag = false;
  }
}

// *****************deal with NaN*************************